pkg/agent/api_docs.go

```diff
@@ -241,6 +243,23 @@ type openAPIDoc struct {
 func newOpenAPI(ctx context.Context, docBytes []byte, baseURL string, prefix string, keepExistingPrefix bool) *openAPIDoc {
 	dlog.Debugf(ctx, "Trying to create new OpenAPI doc: base_url=%q prefix=%q", baseURL, prefix)
 
+	version, err := openAPIVersion(docBytes)
+
+	if err != nil {
+		dlog.Errorln(ctx, "failed to determine open api version:", err)
+		return nil
+	}
+
+	if version == 2 {
+		docBytes, err = convertToOpenAPIV3(docBytes)
+
+		if err != nil {
+			dlog.Errorln(ctx, "failed to convert open api v2 contract to v3:", err)
+			return nil
+
+		}
+	}
+
```

Comment:
It bugs me that this ends up unmarshaling the V2 `docBytes` twice. Would it be cleaner to have

```
v3DocBytes, err := forceOpenAPIV3(docBytes)
```

that checks the version and upgrades if necessary? basically that's smashing `openAPIVersion` and `convertToOpenAPIV3` into a single call, since this looks like the only use for them...

Reply:
done :)